src/Framework/TestCase.php

```diff
@@ -1157,6 +1167,9 @@ public function runBare(): void
     public function setName(string $name): void
     {
         $this->name = $name;
+
+        // Normalized name for dependency resolver
+        $this->providedTests = [\get_class($this) . '::' . $name];
```

Comment:
I do not know if this is relevant but this normalized name may be duplicate between properties, methods and constants because these may have the same name :)

Reply:
This line by itself can indeed overwrite an existing key in the `providedTests`. Fortunately this is part of the contruction logic of new test objects and the `testName` always refers to the `TestCase::`-methods. :relieved: